x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java

```diff
@@ -221,6 +224,11 @@
     protected final boolean transportClient;
     protected final boolean enabled;
     protected final Environment env;
+    private SetOnce<EmailService> emailService = new SetOnce<>();
```

Comment:
what about a `List<NotificationService>` that you fill in `createComponents`? This way you could get rid of the `getReloadableServices()` method entirely and loop through that list in the `reload()` method.

Reply:
7be7b49